File: services/user-service/src/main/java/com/novix/user_service/service/AuthService.java
package com.novix.user_service.service;

import com.novix.user_service.dto.request.LoginRequest;
import com.novix.user_service.dto.request.RegisterRequest;
import com.novix.user_service.dto.response.AuthResponse;

public interface AuthService {
	AuthResponse register(RegisterRequest request);
	AuthResponse login(LoginRequest request);
	AuthResponse refreshToken(String refreshToken);
	void logout(String email);
}

File: frontend/admin/src/services/auth.service.ts
import api from '@/lib/api';
import { AuthResponse, User } from '@/types';

export const authService = {
  async login(email: string, password: string): Promise<AuthResponse> {
    const response = await api.post<AuthResponse>('/api/v1/auth/login', { email, password });
    return response.data;
  },

  async getMe(): Promise<User> {
    const response = await api.get<User>('/api/v1/users/me');
    return response.data;
  },

  saveTokens(accessToken: string, refreshToken: string) {
    localStorage.setItem('admin_accessToken', accessToken);
    localStorage.setItem('admin_refreshToken', refreshToken);
  },

  clearTokens() {
    localStorage.removeItem('admin_accessToken');
    localStorage.removeItem('admin_refreshToken');
  },

  isAuthenticated(): boolean {
    return !!localStorage.getItem('admin_accessToken');
  },

  isAdmin(user: User): boolean {
    return user.roles?.includes('ROLE_ADMIN') || user.roles?.includes('ROLE_CONTENT_MANAGER');
  },
};
